src/replication/DealReplicationService.ts

```diff
@@ -247,6 +253,45 @@ export default class DealReplicationService extends BaseService {
       response.end(JSON.stringify({ id: replicationRequest.id }));
     }
 
+    private async handlePrintCSVReplicationRequest (request: Request, response: Response) {
+      const id = request.params['id'];
+      const replicationRequest = await Datastore.ReplicationRequestModel.findById(id);
+      if (replicationRequest) {
+        const deals = await Datastore.DealStateModel.find({
+          replicationRequestId: id
+        });
+        this.logger.info(`Found ${deals.length} deals from replication request ${id}`);
+        if (deals.length > 0) {
+          const csvRow = [];
+          for (let i = 0; i < deals.length; i++) {
+            const deal = deals[i];
+            csvRow.push({
+              miner_id: deal.provider,
+              deal_cid: deal.dealCid,
+              filename: `${deal.pieceCid}.car`,
+              piece_cid: deal.pieceCid,
+              start_epoch: deal.startEpoch,
+              full_url: `${replicationRequest.urlPrefix}/${deal.pieceCid}.car`
+            });
+          }
+          const csv = new ObjectsToCsv(csvRow);
+          const configDir = config.util.getEnv('NODE_CONFIG_DIR');
```

Comment:
output dir should not be hardcoded here. Either, the request support an output directory/filename, or the response should contain the content of CSV so that the caller can decide which file to save the csv to.